File: src/main/java/zaftnotameni/creatania/util/Actions.java
package zaftnotameni.creatania.util;
import net.minecraft.core.BlockPos;
import net.minecraft.world.entity.LivingEntity;
import net.minecraft.world.entity.item.ItemEntity;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.level.Level;
import zaftnotameni.creatania.registry.CreataniaAdvancements;
import zaftnotameni.creatania.registry.CreataniaItems;
public class Actions {
  public static void killSlimeProduceManagelGrantAchievement(Level level, LivingEntity entity, BlockPos pos) {
    if (level.isClientSide()) return;
    if (!Queries.isSlimeEntity(level, entity)) return;
    ScanArea.forEachPlayerInTheArea(level, pos, 128, p -> true, CreataniaAdvancements.PRODUCE_MANA_GEL_FROM_SLIME::awardTo);
    var stack = new ItemStack(CreataniaItems.MANA_GEL.get(),1);
    entity.kill();
    var itemEntity = new ItemEntity(level, pos.getX(), pos.getY(), pos.getZ(), stack);
    itemEntity.setDeltaMovement(0f, 0.5f, 0f);
    level.addFreshEntity(itemEntity);
  }
}

File: src/main/java/zaftnotameni/creatania/util/Actions.kt
package zaftnotameni.creatania.util

import net.minecraft.core.BlockPos
import net.minecraft.world.entity.LivingEntity
import net.minecraft.world.entity.item.ItemEntity
import net.minecraft.world.entity.player.Player
import net.minecraft.world.item.ItemStack
import net.minecraft.world.level.Level
import zaftnotameni.creatania.registry.Advancements
import zaftnotameni.creatania.registry.Items
import zaftnotameni.creatania.util.Queries.isSlimeEntity
import zaftnotameni.creatania.util.ScanArea.forEachPlayerInTheArea

object Actions {
  @JvmStatic
  fun killSlimeProduceManagelGrantAchievement(level : Level, entity : LivingEntity, pos : BlockPos) {
    if (level.isClientSide()) return
    if (!isSlimeEntity(level, entity)) return
    forEachPlayerInTheArea(level, pos, 128, { p : Player? -> true }) { player : Player? -> Advancements.PRODUCE_MANA_GEL_FROM_SLIME.awardTo(player) }
    val stack = ItemStack(Items.MANA_GEL.get(), 1)
    entity.kill()
    val itemEntity = ItemEntity(level, pos.x.toDouble(), pos.y.toDouble(), pos.z.toDouble(), stack)
    itemEntity.setDeltaMovement(0.0, 0.5, 0.0)
    level.addFreshEntity(itemEntity)
  }
}